### DISCUSSION TITLE
stg ref not working

### DISCUSSION THREAD

**Discussion by shivaprasadbhat** [stacked-git/stgit#462] [Q&A] [answered]
I wasnt sure if this is an issue.

The release notes mentions about the new enhancement "stg new --refresh". I dont see a mention of the previously working "stg ref" instead of "stg refresh" being deprecated.

Is this a bug?

**Comment by jpgrayson** [comment, marked as answer]
In StGit 1.x, unambiguous command prefixes were automatically mapped to the command. For example, with `stg ref`, `ref` is [currently] an unambiguous prefix of `refresh`, and thus was accepted as an alias for `stg refresh`.

In StGit 2.x, command prefixes are no longer automatically mapped to commands. I did this for two reasons. First, it wasn't trivial to do this with the clap library. Second, it avoids the potential for breaking people's implicit aliases by adding new commands.

I recommend creating explicit aliases for any commands that you'd like to abbreviate. E.g.:

`git config --global stgit.alias.ref refresh`

Also note that if using StGit's command line completions for bash, zsh, or fish, `stg ref<tab>` (or whatever the completion expansion key is setup to be) will expand to `stg refresh`.

**Reply by shivaprasadbhat** [reply]
Thanks for answering Pete!
